app/models/ext_management_system.rb

```diff
@@ -503,4 +503,23 @@ def blacklisted_event_names
   def self.blacklisted_events
     BlacklistedEvent.where(:provider_model => name, :ems_id => nil)
   end
+
+  def self.credential_validation(ems, auth_type, creds, revalidate)
```

Comment:
At this point ems hasn't been saved to the database yet so it doesn't have an ems_id.